File: Clg_Placement/Java/_22.ArraySubsetOfAnotherArray.java
class Compute {
    public String isSubset( long a1[], long a2[], long n, long m) {
        HashMap<Long, Integer> mpp = new HashMap<>();
        for(long num: a1){
            mpp.put(num,mpp.getOrDefault(num,0)+1);
        }
        for(long num: a2){
            if(!mpp.containsKey(num) || mpp.get(num)==0) return "No";
            mpp.put(num,mpp.get(num)-1);
        }
        return "Yes";
    }
}

File: Clg_Placement/C++/Arrays/_22.ArraySubsetOfAnotherArray.cpp
string isSubset(int a1[], int a2[], int n, int m) {
    unordered_map<int,int> mpp;
    for(int i=0;i<n;i++){
        mpp[a1[i]]++;
    }
    for(int i=0;i<m;i++){
        if(mpp.find(a2[i])==mpp.end()||mpp[a2[i]]==0) return "No";
        mpp[a2[i]]--;
    }
    return "Yes";
}
